### ISSUE TITLE
[BUG] useModalForm is not working in a list inferencer on any action button

### ISSUE THREAD

**Issue by Jaylord-Bucayu** [refinedev/refine#5558]
### Describe the bug

After I applied useModalForm on my list component and enabling in on a clone and edit button the modal must show and after clicking and clicking the modal is not showing.

### Steps To Reproduce

1. Add useModalForm hook.( Im using antd)
2. To try copy and paste the code in the documentation. https://refine.dev/docs/ui-integrations/ant-design/hooks/use-modal-form/#usage
3. Set a api resource to you useTable hook
4. And try clicking the action button

### Expected behavior

The modal must be filled with the row selected and the modal must be show

### Desktop

Chrome - MAC

**Comment by BatuhanW** [comment]
Hey @Jaylord-Bucayu thanks for the report! We'll check it and get back to you.

**Comment by BatuhanW** [comment]
Hey @Jaylord-Bucayu could you provide a minimal reproducible example?

**Comment by Jaylord-Bucayu** [comment]
Hello I just followed the docs(btw i'm using antd) and what I did is

1. I have a table of clearances it have a fields of (id,amount,status, etc)
2. I have a list inferencer table, the data displays well and looks good.
3. I added a edit button used it in a useModalForm 
const {
        modalProps: editModalProps,
        formProps: editFormProps,
        show: editModalShow,
      } = useModalForm({
        action: "edit",
        warnWhenUnsavedChanges: true,
      });
      
      5. No errors indicated and when clicking the button nothing goes on, no console,network error. plss look my code, many thanks

![sample](https://github.com/refinedev/refine/assets/95568851/c9bb25e6-080b-4a64-aa69-ffac5719e527)
![Screenshot 2024-02-06 at 10 56 47 PM](https://github.com/refinedev/refine/assets/95568851/7bf2137f-918a-46e2-b9c8-cec5b4d5571c)
